### PR TITLE
Adding support to @Remoter interface types

### PR DESCRIPTION
**Pull request by josesamuel**
Added support in Parceler to natively support fields of interface types that are annotated with @Remoter interface. Please see https://josesamuel.com/remoter/ for more details about Remoter

### CODE REVIEW

File: parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.androidtransfuse.util.matcher.Matcher;
+
+/**
+ * Matches a @Remoter inteface
+ *
+ * @author js
+ */
+public class RemoterMatcher implements Matcher<ASTType> {
+
+    private static final ASTType REMOTER = new ASTStringType("remoter.annotations.Remoter");
```
Comment (johncarl81) [inline parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: Why aren't your projects under com.josesamuel matching your groupid on maven central?

Comment (josesamuel) [reply parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: Here is the github https://github.com/josesamuel/remoter. 

I am not sure how to match that with the group id....New to all these.....I thought it is similar to how org.parceler is at  https://github.com/johncarl81/parceler

Comment (johncarl81) [reply parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: I would change the package structure of your project to match the maven group id.  Like this: `com.josesamuel.remoter.annotations.Remoter`.  I'm not sure where I read about this before, and it's certainly not a rule, but maybe just a naming convention.  Found an article here about it: http://blog.sonatype.com/2011/01/maven-tip-project-directories-and-artifact-ids/

Comment (josesamuel) [reply parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: Thanks for the info, i will keep that in mind for future projects. It is too late to change it for remoter, as it is already in use.

Comment (josesamuel) [reply parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: I actually followed the example of Jake Wharton's butter knife https://github.com/JakeWharton/butterknife

Comment (johncarl81) [reply parceler/src/main/java/org/parceler/internal/matcher/RemoterMatcher.java]: Fair enough.

**Comment by johncarl81** [comment]
@josesamuel, do you have an example project I can play with and see this in action?

**Comment by josesamuel** [comment]
@johncarl81 Here is an example project - https://github.com/josesamuel/remoter-parceler-example 

There is a sampleservice and a sampleclient. 

Follow the instructions in the Readme.

It uses the SNAPSHOT version of the parceler with the changes in this pull request.

**Comment by johncarl81** [comment]
@josesamuel I see you added additional commits, are you wanting to include those in this PR?

**Comment by josesamuel** [comment]
No,  nothing more than the files included in the PR.

**Comment by josesamuel** [comment]
I mean just this one commit https://github.com/johncarl81/parceler/commit/a65e6fcc0c1ed2c2a72c64e80e7847e3c85dbf32

**Comment by johncarl81** [comment]
Merged manually.

**Pull request closed without merging**